### PR TITLE
Solved #237

### PR DESCRIPTION
**Pull request by vishwesh3**
Implemented search feature in people and stream drawer.

videotogif_2016 12 12_19 57 12

videotogif_2016 12 12_19 34 24

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@vishwesh3, thanks for the pull request! It looks like you haven't yet signed the Dropbox CLA. Please [sign it here](https://opensource.dropbox.com/cla/).

**Comment by kunall17** [comment]
Looks nice! :+1: 
Thanks for working on this one!

**Comment by Sam1301** [comment]
Hi, @vishwesh3, nice work! I can see you commented the manifestPlaceholders line in build.gradle. I presume its because you were getting gradle sync errors for missing fabricKey.
Here's how I solved it (thanks to niftynei)
1) Create a file named `gradle.properties` in the app directory of the project.
2) Add a line `fabricKey=blah` in it.
3) Sync project with gradle files.

**Comment by vishwesh3** [comment]
@Sam1301 Thanks. Done with gradle build problem.
Please merge this, so that I can contribute on another issues.

**Comment by kunall17** [comment]
@vishwesh3 Maybe, it would be better to remove the border you have given to the whole editText, to something like this 


Rather than [this](https://cloud.githubusercontent.com/assets/12700799/21193587/cf696e92-c252-11e6-96e5-d664985a9603.png),

And can you please 
- Move the gradle commit to another PR as this is not related to this one
- Capitalize the commit messages for better git history!

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java [hunk 1]
```diff
@@ -834,10 +950,20 @@ public Cursor call() throws Exception {
         }
     };
 
+    Callable<Cursor> streamsGeneratorWithFilter = new Callable<Cursor>() {
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: We could achieve the results using one Callable<Cursor> as well, 
If filter editText is not empty then append the WHERE clauses in the SQL query and return the filtered result else no filtering!

**Comment by vishwesh3** [comment]
@kunall17 Yes, `changeCursor()` is working absolutely fine. I have also removed border.
Also appended where clause instead of creating new callable, and Capitalize all commit messages. 
whatsapp image 2016-12-15 at 2 34 10 pm

**Comment by timabbott** [comment]
@kunall17 have you done a final review on this?

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java [hunk 2]
```diff
@@ -327,9 +327,11 @@ protected void onCreate(Bundle savedInstanceState) {
             @Override
             public void onClick(View v) {
                 //set default people list
-                setUpPeopleList(true);
-                //set visibility of this image false
-                ivSearchPeopleCancel.setVisibility(View.GONE);
+                try {
+                    peopleAdapter.changeCursor(getPeopleCursorGenerator().call());
+                } catch (Exception e) {
+                    e.printStackTrace();
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: Use Zlog.LogException(e) wherever Exception is caught!

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java [hunk 3]
```diff
@@ -562,23 +564,67 @@ public void afterTextChanged(Editable s) {
...
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                try {
+                    peopleAdapter.changeCursor(getPeopleCursorGenerator().call());
+                } catch (Exception e) {
+                    e.printStackTrace();
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: Use Zlog.LogException(e) wherever Exception is caught!

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java [hunk 4]
```diff
@@ -307,6 +320,37 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        ivSearchPeopleCancel = (ImageView)findViewById(R.id.iv_people__search_cancel_button);
+        onTextChangeOfPeopleSearchEditText();
+        ivSearchPeopleCancel.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                //set default people list
+                try {
+                    peopleAdapter.changeCursor(getPeopleCursorGenerator().call());
+                } catch (Exception e) {
+                    e.printStackTrace();
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: Use Zlog.LogException(e) wherever Exception is caught!

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java [hunk 5]
```diff
@@ -307,6 +320,37 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        ivSearchStreamCancel = (ImageView)findViewById(R.id.iv_stream_search_cancel_button);
+        onTextChangeOfStreamSearchEditText();
+        ivSearchStreamCancel.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                //set default stream list
+                try {
+                    streamsDrawerAdapter.changeCursor(getSteamCursorGenerator().call());
+                } catch (Exception e) {
+                    e.printStackTrace();
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: Use Zlog.LogException(e) wherever Exception is caught!

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java [hunk 6]
```diff
@@ -539,6 +533,151 @@ public Cursor runQuery(CharSequence charSequence) {
...
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                try {
+                    streamsDrawerAdapter.changeCursor(getSteamCursorGenerator().call());
+                } catch (Exception e) {
+                    e.printStackTrace();
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: Use Zlog.LogException(e) wherever Exception is caught!

**Comment by kunall17** [comment]
@vishwesh3 The code looks much more efficient now .. :+1: 
- Add some bottom border as well, it doesn't look good without the border (I had added the bottom border in the screenshot) you could just add a fake line below the editText to make it look like border
- Rebase the newer commits (after 6feb78fc56de01a605f1533f2bad618f71269db5) and squash them appropiately with the older ones
- You might want to edit the author of the commit's after (or before) 7a1a9761736c8025d75923020269e70fc8ba0084 as they are different
- From the next PR's please format the code as well while writing the code, you can ignore it this time as it would be very tedious task to selective format the code and anyways the formatting will be taken care by #241 

@timabbott after these edit's it would be ready for merge!

**Comment by kunall17** [comment]
Also fix the typo in the commit message in 402df442044b20788d2932dc8110abf75c00619f

**Comment by vishwesh3** [comment]
@kunall17 Please have a final look. Done with all the requested changes.
Added bottom border.
device-2016-12-16-115137

**Comment by kunall17** [comment]
Looks neat now!
- 1 final thing, on pressing the cancel button it would be nice to toggle the soft keyboard and remove the focus from the search edit text in both stream private! 
After that LGTM ! :) :+1:

**Comment by vishwesh3** [comment]
@kunall17 @timabbott Now on cancelling search remove `editText`  focus and hide keyboard.

**Comment by kunall17** [comment]
@vishwesh3 thanks for working on this one! :) 
@timabbott LGTM!

**Comment by timabbott** [comment]
Merged, thanks @vishwesh3 and @kunall17!  It would be awesome for some other folks (e.g. @Sam1301 or @saketkumar95) to do a bit of manual testing to see if we can find any bugs or potential tweaks we should make to this before we do our next release :).

**Comment by saketkumar95** [comment]
@timabbott Yeah sure. :)

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*
				
@vishwesh3, thanks for signing the CLA!

**Comment by Sam1301** [comment]
@vishwesh3 This is a small tweak, the keyboard is visible on the screen even after narrowing to a searched stream/person's private messages.

**Comment by vishwesh3** [comment]
Thanks @Sam1301 I will send new PR soon

**Comment by vishwesh3** [comment]
@Sam1301 created PR #268.

**Pull request merged**